src/test/java/org/zalando/nakadi/domain/StrictJsonParserTest.java

```diff
@@ -35,29 +35,24 @@ public void testVeryComplexJson() throws IOException {
 
     }
 
-    @Test(expected = JSONException.class)
-    public void testBadJson() {
-        StrictJsonParser.parse("{,");
-    }
-
-    @Test(expected = JSONException.class)
-    public void testBadJson2() {
-        StrictJsonParser.parse("{\"name");
-    }
-
-    @Test(expected = JSONException.class)
-    public void testBadJson3() {
-        StrictJsonParser.parse("{name\"}");
-    }
-
-    @Test(expected = JSONException.class)
-    public void testBadJson4() {
-        StrictJsonParser.parse("{\"name\"NaN}");
-    }
-
-    @Test(expected = JSONException.class)
-    public void testBadJson5() {
-        StrictJsonParser.parse("{\"name\":NaN}");
+    @Test
+    public void testInvalidJsonFormats() {
+        final String[] testsData = new String[]{
```

Comment:
this is already invented https://github.com/junit-team/junit4/wiki/parameterized-tests

Reply:
@v-stepanov Do you think that it pays to readability or simplifies development?

Reply:
The problem with your approach is that if you test A, B, C, D, E, F in your test and the test fails for B, it will not even run for C, D, E, F. So you will not have the full picture of which cases fail/successful.

Reply:
I know about this, but honestly, I think that it is not the case here. It either fails or succeeds. But as you insist, I will change it.

Reply:
No, for me it's fine like this, I think it's not a big issue.

Reply:
@v-stepanov too late https://github.com/zalando/nakadi/pull/882/commits/b3058d7174bc3dacafafe29805eeaf5b32ab2ae5